xcube/core/gen/iproc.py

```diff
@@ -260,6 +271,24 @@ def pre_process(self, dataset: xr.Dataset) -> xr.Dataset:
 
         return _normalize_lon_360(dataset)
 
+    def get_spatial_subset(self, dataset: xr.Dataset, dst_region: Tuple[float, float, float, float]) -> xr.Dataset:
+        lon_min, lat_min, lon_max, lat_max = dst_region
+        dataset_subset = dataset.copy()
+        dataset_subset.coords['x'] = xr.DataArray(np.arange(0, dataset.lon.size), dims='lon')
+        dataset_subset.coords['y'] = xr.DataArray(np.arange(0, dataset.lat.size), dims='lat')
+        lon_subset = dataset_subset.lon.where((dataset_subset.lon >= lon_min) & (dataset_subset.lon <= lon_max),
+                                              drop=True)
+        lat_subset = dataset_subset.lat.where((dataset_subset.lat >= lat_min) & (dataset_subset.lat <= lat_max),
+                                              drop=True)
+        x1 = lon_subset.x[0]
```

Comment:
Similarily, make a test, if `dataset` bounds are entirely excluded from `dst_region`, in this case return None and let `gen_cube` ignore the input.